src/ray/raylet/scheduling/policy/composite_scheduling_policy.cc

```diff
@@ -24,27 +24,28 @@ namespace ray {
 namespace raylet_scheduling_policy {
 
 scheduling::NodeID CompositeSchedulingPolicy::Schedule(
-    const ResourceRequest &resource_request, SchedulingOptions options) {
-  switch (options.scheduling_type) {
-  case SchedulingType::SPREAD:
-    return spread_policy_.Schedule(resource_request, options);
-  case SchedulingType::RANDOM:
-    return random_policy_.Schedule(resource_request, options);
-  case SchedulingType::HYBRID:
-    return hybrid_policy_.Schedule(resource_request, options);
-  default:
-    RAY_LOG(FATAL) << "Unsupported scheduling type: "
-                   << static_cast<typename std::underlying_type<SchedulingType>::type>(
-                          options.scheduling_type);
+    const ResourceRequest &resource_request,
+    SchedulingOptions options,
+    SchedulingContext *context /*= nullptr*/) {
+  auto result = Schedule({&resource_request}, options, context);
+  if (result.status.IsSuccess()) {
+    RAY_CHECK(result.selected_nodes.size() == 1);
```

Comment:
This check can be moved to inside the "Schedule" function. if success, check result size == input size

Reply:
> This check can be moved to inside the "Schedule" function. if success, check result size == input size

I don't think so, this is a single schedule method, it is more clear to put `check` in this method directly.